### PR TITLE
[SYCLomatic] Support migration of bar.warp.sync

### PR DESCRIPTION
**Pull request by yihwang-nv**
E2E Test https://github.com/oneapi-src/SYCLomatic-test/pull/641

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 1]
```diff
@@ -409,6 +409,9 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::bits<ExperimentalFeatures>), Exp
         false),
              DPCT_OPT_ENUM("bindless_images", int(ExperimentalFeatures::Exp_BindlessImages),
         "Experimental extension that allows use of bindless images APIs.\n",
+        false),
+             DPCT_OPT_ENUM("non-uniform-groups", int(ExperimentalFeatures::Exp_NonUniformGroups),
+        "Experimental extension that allows non-uniform groups.\n",
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
        "Experimental extension that allows use of non-uniform groups.\n",
```

File: clang/lib/DPCT/AnalysisInfo.h [hunk 2]
```diff
@@ -1223,6 +1223,9 @@ class DpctGlobalInfo {
   static bool useExtBindlessImages() {
     return getUsingExperimental<ExperimentalFeatures::Exp_BindlessImages>();
   }
+  static bool useNonUniformGroups() {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/AnalysisInfo.h]:
```suggestion
  static bool useExpNonUniformGroups() {
```

File: clang/include/clang/DPCT/DPCTOptions.inc (hunk 1 above)
Comment (yihwang-nv) [reply clang/include/clang/DPCT/DPCTOptions.inc]: done

File: clang/lib/DPCT/AnalysisInfo.h (hunk 2 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/AnalysisInfo.h]: done

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**